lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -3605,12 +3605,6 @@ void ByteCodeGenerator::EmitScopeList(ParseNode *pnode, ParseNode *breakOnBodySc
 
                 this->EmitOneFunction(pnode);
                 this->EndEmitFunction(pnode);
-
-                if (paramScope && !paramScope->GetCanMergeWithBodyScope())
```

Comment:
Why are you not "pushing" the paramscope in the StartEmitFunction (line 3579) - then it will be consistent and easy to read.